packages/sanity/src/core/util/draftUtils.ts

```diff
@@ -56,11 +56,27 @@ export function isDraftId(id: string): id is DraftId {
   return id.startsWith(DRAFTS_PREFIX)
 }
 
-/** @internal */
-export function getIdPair(id: string): {draftId: DraftId; publishedId: PublishedId} {
+/**
+ * TODO: Improve return type based on presence of `version` option.
+ *
+ * @internal
+ */
+export function getIdPair(
+  id: string,
+  {version}: {version?: string} = {},
+): {
+  draftId: DraftId
+  publishedId: PublishedId
+  versionId?: string
+} {
   return {
-    draftId: getDraftId(id),
     publishedId: getPublishedId(id),
+    draftId: getDraftId(id),
+    ...(version
+      ? {
+          versionId: id.startsWith(`${version}.`) ? id : [version, getPublishedId(id)].join('.'),
```

Comment:
I think we could benefit from a `getVersionId` function that handles this, which could be reused here https://github.com/sanity-io/sanity/pull/7251/files#diff-e5ae958bd581e7f49548874e1c065e9ac7b325f2d48b753a9bbbd2e4c7649fe8R184

Reply:
I think that would be great.

Reply:
This is being added in a separate PR, or are you adding one already in this PR @juice49 ? :)

Reply:
@RitaDias I wasn't planning to add it here—just so we can draw a line under this branch. However, I do still think it'd be worthwhile. It might even overlap some of your work?

Reply:
@juice49 Yeah the reason why I'm asking is because it would already overlap with my work since I have that on my branch.
That being said if you need it for this branch you can just go ahead and I can adjust my branch?...